src/staking/Staking.ts

```diff
@@ -0,0 +1,54 @@
+import Web3 from 'web3'
+
+export default abstract class Staking {
+  protected readonly web3: Web3
+
+  constructor (protected readonly stakingAddress: string, provider: any) {
```

Comment:
is `stakingAddress` exposed from the outside?

Reply:
changed it to expose